subworkflows/local/enrichment/main.nf

```diff
@@ -31,10 +32,20 @@ workflow ENRICHMENT {
     // ----------------------------------------------------
 
     ch_adjacency
-        .filter { it[0]["enr_method"] == "grea" }
-        .set { ch_adjacency_grea }
+        .map { meta, matrix -> [meta.subMap(["pathway_name"]), meta, matrix] }
+        .join(ch_tools, by: [0])
+        .map {
+            pathway_name, meta, matrix, meta_tools ->
+                def new_meta = meta.clone() + meta_tools.clone()
+                [ new_meta, matrix ]
+            }
+        .branch {
+            grea:  it[0]["enr_method"] == "grea"
+            gsea: it[0]["enr_method"] == "gsea"
```

Comment:
gsea dont use ch_adjacency as input.
So we can maybe just branch ch_tools into grea and gsea at the beginning, and use ch_tools.grea in the GREA section to combine it with ch_adjacency

Reply:
Ok, yes, that sounds good to me. But I'd would change it then when we include gsea.